code/modules/telesci/gps.dm

```diff
@@ -96,7 +100,12 @@ var/list/SPS_list = list()
 	data["location_text"] = get_location_name()
 	var/list/devices = list()
 	if(!emped && transmitting)
-		for(var/obj/item/device/gps/other in gps_list)
+		var/list/ui_list
+		if(view_all)
+			ui_list = all_GPS_list
```

Comment:
instead of a second global list here, why not `ui_list = gps_list + (view_all ? sps_list : null)`

Reply:
Think I did it earlier while coding and it wouldn't work in game when tested.